llama_index/embeddings/openai.py

```diff
@@ -304,6 +304,14 @@ def __init__(
         self._aclient = None
         self._http_client = http_client
 
+    @validator("api_key", pre=True)
+    def _validate_api_key(cls, v: Optional[str]) -> Optional[str]:
```

Comment:
is this function being used somewhere

Reply:
I think no

Reply:
@hatianzhang the decorator hooks it into pydantic, it will be called automatically 👍🏻

Reply:
oh crap -- I should not have merged this 😓 This will break azure